online-shop/src/app/auth.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { DataService } from './data.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+  constructor(private dataService : DataService, private router:Router){}
+  
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): boolean {
+      if(this.dataService.getCurrentUser().username != '' && this.dataService.getCurrentUser().fullName != null && this.dataService.getCurrentUser().roles != null)
```

Comment:
it's not mandatory to explicitly compare with null/'' (empty string). Condition can be simplified (easy to read):
`if(this.dataService.getCurrentUser().username && this.dataService.getCurrentUser().fullName && this.dataService.getCurrentUser().roles)`
if condition will evaluate to true if value is not: null/undefined/empty_string/0/false